File: Golang/SimplifyPath/SimplifyPath_test.go
package SimplifyPath

import "testing"

func TestSimplifyPath(t *testing.T) {

	if simplifyPath("/home/") == "/home" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if simplifyPath("/a/./b/../../c/") == "/c" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if simplifyPath("/../") == "/" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if simplifyPath("/home//foo/") == "/home/foo" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if simplifyPath("/a/b/c") == "/a/b/c" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SimplifyPathTest.py
from unittest import TestCase

from SimplifyPath import SimplifyPath


class TestSimplifyPath(TestCase):
    def test_simplifyPath(self):
        sp = SimplifyPath()

        self.assertEqual(sp.simplifyPath("/home/"), "/home")

        self.assertEqual(sp.simplifyPath("/a/./b/../../c/"), "/c")

        self.assertEqual(sp.simplifyPath("/../"), "/")

        self.assertEqual(sp.simplifyPath("/home/foo/"), "/home/foo")

        self.assertEqual(sp.simplifyPath("/a/b/c"), "/a/b/c")
